### PR TITLE
PPTP-1519 - re-hydrate a Registration model from SubscriptionDisplayResponse

### PR DESCRIPTION
**Pull request by SteveSugden**
Deals with all currently supported org types except partnerships (general partnership is assumed)

- code-coverage temporarily reduced to 94%

### CODE REVIEW

File: app/uk/gov/hmrc/plasticpackagingtaxregistration/models/Registration.scala [hunk 1]
```diff
@@ -49,4 +52,81 @@ object Registration {
...
+      ).map(OrgType.withName)
+
+    val incorporationDetails = organisationType match {
+      case Some(OrgType.UK_COMPANY) =>
+        Some(
+          IncorporationDetails(
+            companyNumber = subscription.legalEntityDetails.customerIdentification1,
+            companyName = subscription.legalEntityDetails.customerDetails.organisationDetails.map(
+              _.organisationName
+            ).getOrElse("UNKNOWN"),
```
Comment (RyanKellyHMRC) [inline app/uk/gov/hmrc/plasticpackagingtaxregistration/models/Registration.scala:86]: I'm wondering whether we'd be better served throwing an IllegalStateException here? We shouldn't ever see this situation should we?

Comment (RyanKellyHMRC) [reply app/uk/gov/hmrc/plasticpackagingtaxregistration/models/Registration.scala:86]: And we wouldn't ever want to present "UNKNOWN" to the user.

File: app/uk/gov/hmrc/plasticpackagingtaxregistration/models/Registration.scala [hunk 2]
```diff
@@ -49,4 +52,81 @@ object Registration {
...
+            ctutr = subscription.legalEntityDetails.customerIdentification2.getOrElse("UNKNOWN"),
+            businessVerificationStatus = "UNKNOWN",
+            companyAddress = IncorporationAddressDetails(),
+            registration =
+              IncorporationRegistrationDetails(registrationStatus = "UNKNOWN",
+                                               registeredBusinessPartnerId = None
```
Comment (RyanKellyHMRC) [inline app/uk/gov/hmrc/plasticpackagingtaxregistration/models/Registration.scala:92]: I think we can live with these other UNKNOWNs though. Perhaps we ought to refactor Registration to make them optional, although that also feels "messy"?

**Review by RyanKellyHMRC — COMMENTED** [review]
Looks ok. I'm not totally sold on the decision (I'm aware that this was my proposal 😁) to convert to Registration rather than chucking a SubscriptionDisplayResponse (although not sure I can live with that name) back to the FE. I wonder whether we need a FE spike to look at how much would need to change if we were to do that?

**Review by paulblakehmrc — APPROVED** [review]

**Pull request merged**